### PR TITLE
better transaction management

### PR DESCRIPTION
**Pull request by eringerli**
This uses two new methods to do the transaction management.

### CODE REVIEW

File: Adafruit_SPIDevice.cpp [hunk 1]
```diff
@@ -83,10 +83,7 @@ Adafruit_SPIDevice::~Adafruit_SPIDevice() { delete _spiSetting; }
  * init
  */
 bool Adafruit_SPIDevice::begin(void) {
-  if (_cs != -1) {
-    pinMode(_cs, OUTPUT);
```
Comment (ladyada) [inline Adafruit_SPIDevice.cpp:87]: removing this line is not optional

**Review by ladyada — CHANGES_REQUESTED** [review]

File: Adafruit_SPIDevice.cpp (hunk 1 above)
Comment (eringerli) [reply Adafruit_SPIDevice.cpp:87]: OK, oversaw `pinMode()`

Comment (eringerli) [reply Adafruit_SPIDevice.cpp]: I removed that commit from the PR

**Comment by eringerli** [comment]
wait a minute, I'll rebase

**Comment by eringerli** [comment]
done

**Review by ladyada — APPROVED** [review]

**Comment by ladyada** [comment]
thanks! as you can see, its very easy to slip on a single line, that would cause the entire library to stop working :)

**Pull request merged**